File: 08-Colas_de_prioridad_acotada/cpp/node.cpp
#include "node.h"

template <typename T>
Node<T>::Node() {
    next = nullptr;
}

template <typename T>
Node<T>::Node(T data) {
    this->data = data;
    next = nullptr;
}

template <typename T>
Node<T>::Node(T data, Node<T> *next) {
    this->data = data;
    this->next = next;
}

template <typename T>
Node<T>::~Node() {
    delete next;
}

template <typename T>
T Node<T>::getData() {
    return data;
}

template <typename T>
void Node<T>::setData(T data) {
    this->data = data;
}

template <typename T>
Node<T>* Node<T>::getNext() {
    return next;
}

template <typename T>
void Node<T>::setNext(Node<T> *next) {
    this->next = next;
}

File: 08-Colas_de_prioridad_acotada/python/node.py
class Node:
    data = None
    next = None

    def __init__(self, data = None, next=None):
        self.data = data
        self.next = next

    def getData(self):
        return self.data

    def setData(self, data):
        self.data = data

    def getNext(self):
        return self.next

    def setNext(self, next):
        self.next = next

    def __str__(self):
        return "|" + str(self.data) + "| --> "

    def __repr__(self):
        return str(self.data)
